features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/OtelTracerBuilderProviderTest.kt

```diff
@@ -446,4 +421,155 @@ internal class OtelTracerBuilderProviderTest {
+        val tracer = testedOtelTracerProviderBuilder
+            .setPartialFlushThreshold(1)
+            .setSampleRate(0.0)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flushAndClose()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isEqualTo(-1)
+    }
+
+    @Test
+    fun `M use user-keep or user-not-keep priority W buildSpan { provided random sample rate }`(
+        @DoubleForgery(min = 0.0, max = 100.0) sampleRate: Double
+    ) {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setPartialFlushThreshold(1)
+            .setSampleRate(sampleRate)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flushAndClose()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isIn(-1, 2)
+    }
+
+    @Test
+    fun `M use keep priority W buildSpan { not provided sample rate }`() {
+        // Given
+        val tracer = testedOtelTracerProviderBuilder
+            .setPartialFlushThreshold(1)
+            .build()
+            .tracerBuilder(fakeInstrumentationName)
+            .build()
+
+        // When
+        val span = tracer
+            .spanBuilder(fakeOperationName)
+            .startSpan()
+        span.end()
+        val spanContext = span.spanContext as OtelSpanContext
+        val ddSpanContext = spanContext.delegate as DDSpanContext
+
+        // Then
+        // make sure we are flushing the spans in order to set the sampling priority
+        tracer.flushAndClose()
+        val priority = ddSpanContext.samplingPriority
+        assertThat(priority).isEqualTo(1)
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun Tracer.flushAndClose() {
+        val coreTracer: CoreTracer = getFieldValue("tracer")
+        coreTracer.flush()
+        val pendingTraceBuffer: PendingTraceBuffer = coreTracer.getFieldValue("pendingTraceBuffer")
+        val pendingTraceBufferWorker: Thread = pendingTraceBuffer.getFieldValue("worker")
+        try {
+            pendingTraceBufferWorker.join(10000)
+        } catch (e: InterruptedException) {
+            // ignore
```

Comment:
should we not log that this has happened? Alternatively if we keep it, I have some memory there is a convention to change the variable name to `ignored` in the catch block